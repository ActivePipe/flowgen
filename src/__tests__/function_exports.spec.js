import compiler from "../cli/compiler";
import beautify from "../cli/beautifier";

it("should handle exported es module functions", () => {
  const ts = `export function routerReducer(state?: RouterState, action?: Action): RouterState;
export function syncHistoryWithStore(history: History, store: Store<any>, options?: SyncHistoryWithStoreOptions): History & HistoryUnsubscribe;
`;
  const result = compiler.compileDefinitionString(ts);
  expect(beautify(result)).toMatchSnapshot();
});
