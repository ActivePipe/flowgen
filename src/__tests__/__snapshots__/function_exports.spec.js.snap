// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should handle exported es module functions 1`] = `
"declare export function routerReducer(
  state?: RouterState,
  action?: Action
): RouterState;
declare export function syncHistoryWithStore(
  history: History,
  store: Store<any>,
  options?: SyncHistoryWithStoreOptions
): History & HistoryUnsubscribe;
"
`;
